pagefaultgames/pokerogue
# [QoL] [FIXED!] Pokemon Info Container always on top of Enemy Modifier Bar

## **IMPORTANT**
**This PR got recently merged and got reverted, because I missed the scenario when you already have a full team and then catch a pokemon and are asked which pokemon you want to replace (very sorry about that).** 

**The previous bug**
![image](https://github.com/pagefaultgames/pokerogue/assets/1838682/036e4abc-3983-4e8f-8087-198268ea5393)

**I discarded my fiddling with `setDepth` on the `parentContainer ` because it's too prone to cause weird UI issues. Now I keep the order/depth as it is and just hide the `enemyModifierBar` whenever the `pokemonInfoContainer` is displayed. This is way safer.**


.
.
.



## What are the changes?
When catching a pokemon, the `enemyModifierBar` overlaps the `pokemonInfoContainer`. See Videos for before vs after. 

## Why am I doing these changes?
This is neither visually pleasing nor intended. And I've found no other PR fixing this issue. 

## What did change?
- added `showEnemyModifierBar` and `hideEnemyModifierBar` to allow outside access to the visibilitay of the bar
- hide the bar when the `pokemonInfoContainer` is shown
- show the bar when the `pokemonInfoContainer` is hidden (this is not visually noticable since after you catch a pokemon there is no pokemon anymore that could have any modifiers)

### Screenshots/Videos

**1. BEFORE**

https://github.com/pagefaultgames/pokerogue/assets/1838682/e31aa360-abcd-4bf3-a7d0-0457550c2ddc


**2. AFTER (THIS TIME WITH THE SCENARIO THAT CAUSED THE REVERT, REPLACING A POKEMON)** 

https://github.com/pagefaultgames/pokerogue/assets/1838682/c9ed182b-0515-49b5-97c1-e4fd20105614


**3. EGG HATCHING PHASE (NO EFFECT)**

https://github.com/pagefaultgames/pokerogue/assets/1838682/b08add73-cf86-48ae-98bf-764052089b2c


## How to test the changes?
Overrides:
Give you a master ball:
`export const POKEBALL_OVERRIDE: { active: boolean, pokeballs: PokeballCounts } = {
  active: true,
  pokeballs: {
    [PokeballType.POKEBALL]: 5,
    [PokeballType.GREAT_BALL]: 0,
    [PokeballType.ULTRA_BALL]: 0,
    [PokeballType.ROGUE_BALL]: 0,
    [PokeballType.MASTER_BALL]: 1,
  }
};`

Give the enemy a held item:
`export const OPP_HELD_ITEMS_OVERRIDE: Array<ModifierOverride> = [{name: "BERRY", count: 1, type: BerryType.SITRUS}];`

Start a new run and catch the first pokemon with the master ball. The Info Container will open and is displayed ABOVE the held berry.

**EDIT: Also test the case when you have a full team, catch a pokemon and try to replace it.**

## Checklist
